posts/hands-on-guide-to-migrating-postgres-with-pglogical-and-aws-dms-part-2.md

```diff
@@ -0,0 +1,232 @@
+
+At the very bottom of the endpoint creation screen is a checkbox that says `Use endpoint connection attributes`. Click this checkbox and a text field will appear. Add the following text to the text field *Extra endpoint connection attributes*:
+```bash
+PluginName=PGLOGICAL;slotName=dms_slot_01
+```
+
+Test the connection and then press save. Repeat this process for all of the replication slots that we created in Phase I of the migration.
+
+Now that we have source endpoints for all of the replication slots that exist on the source database, we need to create a target endpoint. Follow the same process as above but instead specify the connection string for your RDS instance and skip the bit about using "extra endpoint connection attributes".
+
+### Step 3 - Create the DMS Tasks
+
+In order to setup our DMS Tasks, we will navigate to the DMS console and click on "Create a DMS Task". The absolute most critical step here, is to select "Replicate data changes only". We do not want to direct DMS to perform full load, as we have already done this step outside of DMS. Specify the Source Endpoint for your first "Replication Slot Endpoint" and choose the target database (RDS endpoint). This setup part, and the following part, will be the same across all of the DMS tasks you create.
+
+One of the next settings is: `CDC Start mode for source transactions` and there is an option for `Custom CDC start point for source transactions`. Choose `Specify a log sequence number`. In the `System change number` text field, input the log sequence number that we ascertained from `Step 6 - Capture the Log Sequence Number` in Part 1 of this series. This critical step will tell DMS where it should "pick up" the change data capture on the Source Database.
+
+The only other settings that will be the same across all DMS tasks are as follows:
+- do create recovery table on target DB
+- do nothing - target table prep (we already handled target table prep before full load)
+- turn on cloudwatch and select all logs
+- create recovery table in `public` schema (or what ever schema name you use)
+- enable all awsdms control tables
+
+Here is where things change based on which DMS Task, and therefore which replication slot, and therefore which tables we are considering...
+
+If the tables that this task references has the tables that `Have a primary key and a separate unique constraint`, make sure `batch-optimized apply` is disabled. For all other tasks (replication slots), it's okay to turn this on.
+
+If the tables that this task references have any tables that are included in your `Large Object` or `LOB` list that we compiled above, then we have specific settings to take care of. Set the large object settings to `Limited LOB Mode` and put in a value that makes sense for the "maximum lob size" based off of your list. If the DMS Task has a table where some truncations are not okay, then we have some more special settings to add below. 
```

Comment:
"If the tables that this task references ( REMOVE have any tables that )are included in your...